test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -84,9 +84,6 @@ void ProcessExport(Batch<Metric> batch)
             // giving some breezing room for the End callback to complete
             await Task.Delay(TimeSpan.FromSeconds(1));
 
-            // Invokes the TestExporter which will invoke ProcessExport
-            metricReader.Collect();
```

Comment:
just confirming:
if we remove explicit Collect(), then we are relying on the provider dispose... which has a hard-coded 5000 msec to finish flushing.
On the other hand, if we keep explicit Collect(), we can pass an explicit timeout to it. (we were not doing it now and was relying on Infinite timeout).

Net effect of this change in test code:
Previously we would wait indefinitely for Collect() to be over.
With this PR, we hope Collect() will be over in 5 secs.

If we ever need more than 5 sec, we need to bring back Collect(maxTimeInMilliSecs..).

Reply:
Yes. If we need more time, what we need to do in the test cases:
1. never rely on provider Dispose for validation.
2. always use Collect explicitly, and validate the result after Collect.
3. after all the validation is done, use provider Dispose to release the resources.